open-craft/opencraft
# [BB-1069] Add Privacy Policy and Terms of Use

* Add a checkbox to the registration form to indicate acceptance of the Privacy Policy, without which the application will not be allowed.
* Add templates for the Privacy Policy and Terms of Use, and link these from the descriptions of the relevant registration checkboxes.
* Add a field to the `Instance` model indicating the timestamp at which the Privacy Policy was accepted.
* Populate this field with the timestamp of the `BetaTestApplication` creation when creating an `Instance` in response to email validation, which is effectively the time at which we can be certain the Privacy Policy was accepted.

## Testing Instructions

* Merge this branch to your local devstack and apply migrations.
* Navigate to the registration form and click the links for the Privacy Policy and Terms of Use.
* Attempt to submit without accepting the Privacy Policy.
  * The submission should be denied, and an error message displayed.
* Accept the privacy policy and resubmit.
  * The submission should proceed, and you will see that it is under review.
* In the DJango admin, locate the application and set its status to `Approved`.
* In the Django console output should be an email to validate your email address; logout of Django admin and then navigate to the link in the email.
* In the Django admin, observe the newly-created OpenedX Instance.
* Navigate to the instance and observe the `Accepted Privacy Policy` field.
